pkg/operator/controller/garden/reconciler_reconcile.go

```diff
@@ -103,37 +87,72 @@ func (r *Reconciler) reconcile(
 		return reconcile.Result{}, err
 	}
 
-	log.Info("Deploying and waiting for gardener-resource-manager to be healthy")
+	log.Info("Instantiating component deployers")
+	vpaCRD := vpa.NewCRD(applier, nil)
+	hvpaCRD := hvpa.NewCRD(applier)
 	gardenerResourceManager, err := r.newGardenerResourceManager(garden, secretsManager)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	if err := component.OpWaiter(gardenerResourceManager).Deploy(ctx); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	log.Info("Reconciling system resources")
-	if err := r.newSystem().Deploy(ctx); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	log.Info("Reconciling VPA components")
+	system := r.newSystem()
 	verticalPodAutoscaler, err := r.newVerticalPodAutoscaler(garden, secretsManager)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	if err := verticalPodAutoscaler.Deploy(ctx); err != nil {
-		return reconcile.Result{}, err
-	}
-
-	log.Info("Reconciling HVPA controller")
 	hvpaController, err := r.newHVPA()
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	if err := hvpaController.Deploy(ctx); err != nil {
+	etcdDruid, err := r.newEtcdDruid()
+	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	var (
+		g            = flow.NewGraph("Garden reconciliation")
+		deployVPACRD = g.Add(flow.Task{
+			Name: "Deploying custom resource definition for VPA",
+			Fn:   flow.TaskFn(vpaCRD.Deploy).DoIf(vpaEnabled(garden.Spec.RuntimeCluster.Settings)),
+		})
+		reconcileHVPACRD = g.Add(flow.Task{
+			Name: "Reconciling custom resource definition for HVPA",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if !hvpaEnabled() {
+					return hvpaCRD.Destroy(ctx)
+				}
+				return hvpaCRD.Deploy(ctx)
```

Comment:
Could you use `component.OpDestroyer` on instantiation of the component instead of the `if` clause inside the task?